osquery/core/system.cpp

```diff
@@ -131,7 +131,13 @@ Status checkStalePid(const std::string& content) {
       return Status(1, "Error querying processes: " + q.getMessageString());
     }
 
-    if (q.rows().size() >= 1 && q.rows().front()["name"] == "osqueryd") {
+    bool is_osqueryd = false;
+    if (q.rows().size() >= 1) {
```

Comment:
```cpp
if (q.rows().size() >= 1 && 
    q.rows().front().count("name") > 0 && 
    q.rows().front().at("name") == "osqueryd") {
  // If you want to avoid setting a flag var.
}
```

Would it be worth deriving vector for `QueryData` and adding `anyMatches(columnName, expression)` so `q.anyMatches("name", "osqueryd")` could be a thing?